src/nunavut/lang/c/templates/_init_macros.j2

```diff
@@ -1,21 +1,29 @@
-{%- macro init_field(instance_ptr_token, field, index='') -%}
-    {%- if field is not padding -%}
-    {%- if field.data_type is CompositeType -%}
-        {{ field.data_type | declaration }}_init(&({{ instance_ptr_token }}->{{ field | id }}{{ index }}));
-    {%- elif field.data_type is FixedLengthArrayType -%}
-        {%- if field.data_type.element_type is PrimitiveType -%}
-        memset(&({{ instance_ptr_token }}->{{ field | id }}[0]), 0, sizeof({{ field.data_type | declaration }}) * {{ field.data_type.capacity }});
+{%- macro init_field(field, data_type=None, ref='out_instance->') -%}
+    {%- if field is None -%}
```

Comment:
Huh. I thought this didn't work in Jinja : https://jinja.palletsprojects.com/en/2.11.x/templates/#assignments ? Are you sure this is working like you expect it to? Do we need to use a namespace object?